test/spec/modules/conversantAnalyticsAdapter_spec.js

```diff
@@ -0,0 +1,733 @@
+
+      let payload = caObj.createPayload(REQUEST_TYPE, AUCTION_ID);
+      expect(payload.requestType).to.equal(REQUEST_TYPE);
+      expect(payload.auction.auctionId).to.equal(AUCTION_ID);
+      expect(payload.auction.preBidVersion).to.equal(PREBID_VERSION);
+      expect(payload.auction.sid).to.equal(VALID_ALWAYS_SAMPLE_CONFIG.options.site_id);
+      expect(Object.keys(payload.auction).length).to.equal(3);
+
+      expect(typeof payload.adUnits).to.equal('object');
+      expect(Object.keys(payload.adUnits).length).to.equal(0);
+
+      expect(Object.keys(payload).length).to.equal(3);
+
+      conversantAnalytics.disableAnalytics();
+    });
+
+    it('cleanCache() should purge old objects and not new ones', function() {
+      const CURR_TIME = Date.now();
+      const EXPIRED_TIME = CURR_TIME - caObj.MAX_MILLISECONDS_IN_CACHE;
+      let cacheObj = {};
+      cacheObj.foo = { timeReceived: CURR_TIME };
+      cacheObj.bar = { timeReceived: EXPIRED_TIME };
+
+      caObj.cleanCache(cacheObj, CURR_TIME);
+
+      expect(Object.keys(cacheObj).length).to.equal(1);
+      expect(cacheObj.foo).to.not.be.undefined;
+      expect(cacheObj.bar).to.be.undefined;
+    });
+  });
+
+  describe('Bid Timeout Event Tests', function() {
+    const bidTimeoutPayload = [{
+      'bidId': '80882409358b8a8',
+      'bidder': 'conversant',
+      'adUnitCode': 'MedRect',
+      'auctionId': 'afbd6e0b-e45b-46ab-87bf-c0bac0cb8881'
+    }, {
+      'bidId': '9da4c107a6f24c8',
+      'bidder': 'conversant',
+      'adUnitCode': 'Leaderboard',
+      'auctionId': 'afbd6e0b-e45b-46ab-87bf-c0bac0cb8881'
+    }
+    ];
+
+    beforeEach(function () {
+      conversantAnalytics.enableAnalytics(VALID_ALWAYS_SAMPLE_CONFIG);
+    });
+
+    afterEach(function () {
+      conversantAnalytics.disableAnalytics();
+    });
+
+    it('should put both items in timeout cache', function() {
+      expect(Object.keys(caObj.timeoutCache).length).to.equal(0);
+      events.emit(constants.EVENTS.BID_TIMEOUT, bidTimeoutPayload);
+      expect(Object.keys(caObj.timeoutCache).length).to.equal(2);
+      expect(requests.length).to.equal(0);
+    });
+  });
+
+  describe('Render Failed Tests', function() {
+    const RENDER_FAILED_PAYLOAD = {
+      reason: 'reason',
+      message: 'value',
+      adId: '57e03aeafd83a68'
+    };
+
+    const RENDER_FAILED_PAYLOAD_NO_ADID = {
+      reason: 'reason',
+      message: 'value'
+    };
+
+    beforeEach(function () {
+      conversantAnalytics.enableAnalytics(VALID_ALWAYS_SAMPLE_CONFIG);
+    });
+
+    afterEach(function () {
+      conversantAnalytics.disableAnalytics();
+    });
+
+    it('should empty adIdLookup and send data', function() {
+      caObj.adIdLookup[RENDER_FAILED_PAYLOAD.adId] = {
+        bidderCode: 'bidderCode',
+        adUnitCode: 'adUnitCode',
+        auctionId: 'auctionId',
+        timeReceived: Date.now()
+      };
+
+      expect(Object.keys(caObj.adIdLookup).length).to.equal(1);
```

Comment:
Not needed.  It has just been created.

Reply:
Because it's a global variable I want to make sure that it is set correctly before testing.  Part of good testing is to ensure that your pre-conditions are true and tested.